src/app/utils/api.class.ts

```diff
@@ -0,0 +1,33 @@
+import axios, { type AxiosInstance, type AxiosRequestConfig } from 'axios';
+
+const instance = axios.create({
+  baseURL: 'http://localhost:3000/api/v1',
+});
+
+interface IApiProps {
+  url: string;
+  data: object;
+  params?: AxiosRequestConfig<any>;
+}
+
+class Api {
+  constructor(protected readonly request: AxiosInstance) {}
+
+  create({ url, data, params = {} }: IApiProps) {
+    return this.request.post(url, data, params);
+  }
+
+  find({ url, params = {} }: Omit<IApiProps, 'data'>) {
+    return this.request.get(url, { params });
+  }
+
+  update({ url, data, params = {} }: IApiProps) {
+    return this.request.put(url, data, params);
+  }
+
+  delete({ url, params = {} }: Omit<IApiProps, 'data'>) {
+    return this.request.delete(url, params);
+  }
+}
+
+export const apiClient = new Api(instance);
```

Comment:
Should we not define an interceptor that populates the bearer token before sending a request?